src/hooks/index.js

```diff
@@ -348,3 +348,93 @@ exports.checkSchoolOwnership = hook => {
 				throw new errors.Forbidden('You do not have valid permissions to access this.');
 		});
 };
+
+//TODO: later: Template building
+//z.B.: maildata.template = { path: "../views/template/mail_new-problem.hbs", "data": { "firstName": "Hannes", .... } };
+//if (maildata.template) { [Template-Build (view client/controller/administration.js)] }
+// mail.html = generatedHtml || "";
+exports.sendEmail = (hook, maildata) => {
+	const userService = hook.app.service('/users');
+	const mailService = hook.app.service('/mails');
+	
+	let roles = (typeof maildata.roles === "string" ? [maildata.roles] : maildata.roles) || [];
+	let emails = (typeof maildata.emails === "string" ? [maildata.emails] : maildata.emails) || [];
+	let userIDs = (typeof maildata.userIds === "string" ? [maildata.userIds] : maildata.userIds) || [];
+	let receipients = [];
+	
+	let promises = [];
+
+	if (roles.length > 0) {
+		promises.push(
+			userService.find({query: {
+				roles: [roles],
+				schoolId: hook.data.schoolId,
+				$populate: ['roles']
+			}})
+		);
+	}
+	
+	if (userIDs.length > 0){
+		userIDs.map (id => {
+			promises.push(
+				userService.find({query: {
+					_id: mongoose.Types.ObjectId(id)
+				}})
+			);
+		});
+	}
+	
+	if (emails.length > 0){
+		emails.map(email => {
+			let re = /\S+@\S+\.\S+/;
+    		if (re.test(email)){
+				receipients.push(email);
+			}
+		});
+	}
+
+	if(promises.length > 0){
```

Comment:
maybe just an idea which you could try out is to get rid of the if here and the duplicate code further on.
Just a bit of thinking now:
if promise.length > 0 then everything works fine and otherwise promise is an empty array.
Promise.all on an empty array should return an empty array.
Meaning you could get rid of the if and just deal with the result in the "then" case, where you would check if it is empty if so send the mails otherwise do your checks as you have them already and then send the mails.

Reply:
I tried to do it but it didn't work for me; if the promise array is empty it doesn't trigger the "then"...